_posts/2019-03-28-introducing-d2.md

```diff
@@ -0,0 +1,83 @@
+
+4. `d2 style commit check "feat: this is a test"` - check whether or not a string matches the [conventional commits](http://conventionalcommits.org/) specification (this is handy [in a git hook](https://github.com/dhis2/cli/blob/master/package.json#L19)!)
+
+5. `d2 create cli my-cli-module` - Create a new CLI module. This module can be published and run stand-alone, or included as a sub-module of `d2`
+
+6. Explore! You can run `d2 --help` to list all the commands and namespaces available, then `d2 style --help` and all the way down to `d2 style js install --help`.
+
+## How do you install it?
+
+The pre-requisites for `d2` are [nodejs](https://nodejs.org/en/download/) and the package manager [yarn](https://yarnpkg.com/lang/en/docs/install/).
+
+```sh
+> yarn global add @dhis2/cli
+> d2 --help
+d2 <command>
+
+Commands:
+  d2 app                   Front-end application and library commands
+  d2 cluster               Manage DHIS2 Docker clusters             [aliases: c]
+  d2 create <type> [name]  Create various DHIS2 components from templates
+  d2 style                 DHIS2 programmatic style for commit msgs/code
+                                                                    [aliases: s]
+  d2 utils                 Utils for miscellaneous operations
+  d2 debug                 Debug local d2 installation
+
+Options:
+  --version   Show version number                                      [boolean]
+  --config    Path to JSON config file
+  -h, --help  Show help                                                [boolean]
+```
+
+If you just want to run a single command without installing `d2` globally, you can use `npx`, for instance both commands above are equivalent to `npx @dhis2/cli --help`
+
+**NB** installation with `npm install -g @dhis2/cli` should also work but yarn is recommended.
+
+**NB** there is an open issue tracking support for zero-dependency binary executables (which wouldn't require `node` be pre-installed).
+
+If either installing `d2` with `npm` or just by downloading an executable appeals to you, please indicate your support on [issue #34](https://github.com/dhis2/cli/issues/34) in the `cli` repository
+
+## What about other command-line utilites?
+
+The code-style and packages tools introduced by Viktor [in an earlier post](/blog/2018/12/07/packages-and-conventions.html) are now deprecated in favor of their `d2` equivalents (`d2 style` instead of `code-style`. Yarn workspaces should be used instead of `packages` for simple mono-repo support, and `d2 utils release` replaces `packages release`, although this should only need to run in CI environments like [travis](https://www.travis-ci.com))
```

Comment:
`@`-usernames on github works too 👌 

The one thing `packages` offers that d2 does not is running commands on multiple sub-packages (e.g. `packages exec yarn build`).

Reply:
👍 updated